src/true_peak.rs

```diff
@@ -41,6 +41,12 @@ enum UpsamplingScanner {
     Generic4F(Box<[Interp4F<[f32; 1]>]>),
 }
 
+#[derive(Debug)]
+struct UpsamplingScanner {
+    mode: UpsamplingScannerMode,
+    processed: usize,
```

Comment:
It doesn't look like you actually increment this anywhere. Also this would overflow after just 4 billion samples on 32 bit platforms, and then the subtraction below would become `0` again shortly so it would be best to make sure that this doesn't happen

Reply:
Good catch! We're using this crate in an application where we require 64-bit so I'm a bit blinded by context :)